braindecode/datasets/base.py

```diff
@@ -141,6 +163,28 @@ def transform(self, value):
             raise ValueError('Transform needs to be a callable.')
         self._transform = value
 
+    @property
+    def description(self):
+        return self._description
+
+    def set_description(self, description, overwrite=False):
+        """Update (add or overwrite) the dataset description.
+
+        Parameters
```

Comment:
is this code same as in BaseDataset? then atm it is inherited anyways right?

Reply:
Yes, code is the same and currently inherited. Same goes for the `transform` stuff. So either we inherit and remove all code duplicates or don't inherit and have the code in both classes.

Reply:
Earlier the tendency was to not inherit. So I will keep the code?

Reply:
Atm I would tend towards making 3 classes:
1. BaseDataset (common functionality, doesn't get any MNE object passed, __getitem__ raises NotImplementedError)
2. WindowsDataset 
3. ContinousDataset (or RawDataset, so gets raws)
Then we could prevent duplicating code and structure would be clean to understand I think.
What do you think?

Reply:
Since it may create some conflicts with other PR https://github.com/braindecode/braindecode/pulls , we could also do this later in a separate PR